### ISSUE TITLE
Adding Advection Schemes 

### ISSUE THREAD

**Issue by anthony-meza** [JuliaClimate/MeshArrays.jl#120]
Leaving this note here for myself. I think it would be nice to implement some of the MITgcm advection schemes [pkg/generic_advdiff](https://github.com/MITgcm/MITgcm/tree/0c90ef582a21389381df30a790a60326006fd53f/pkg/generic_advdiff). 

In particular, I am interested in implementing things like [this](https://github.com/MITgcm/MITgcm/blob/0c90ef582a21389381df30a790a60326006fd53f/pkg/generic_advdiff/gad_u3_adv_r.F)

**Comment by gaelforget** [comment]
Agreed. Maybe we could start by porting this (below) to MeshArrays.jl ?

https://github.com/JuliaOcean/PlanktonIndividuals.jl/tree/614223547c87a85eab030125ee8bf8e5f0d347fb/src/Biogeochemistry/Advection

**Comment by gaelforget** [comment]
could be a good topic for [ECCO hackweek 2024](https://ecco-hackweek.github.io/ecco-2024/), next week, maybe?

**Comment by anthony-meza** [comment]
I think so! To clarify my initial issue (because it was very vague), I was interested in computing advective flux convergences using the MITgcm diagnostics of temperature and velocity. My thought was that it would be nice if a function existed such that ``advective_flux convergence(temperature, velocity, advection_scheme = "third_order_upwind bias_advection"``.  But you should take it in whatever direction you like :)
